src/extension.ts

```diff
@@ -53,6 +55,10 @@ export async function activate(context: ExtensionContext) {
       : DEFAULT_SEARCH_CRITERIA;
   const workspaceFolders = workspace.workspaceFolders;
 
+  // When a workspace is first opened and already has an open document, the setDecoration method has to be called twice.
+  // If it is isn't, the user will have to tab between documents before the decorations will render.
+  let setDecorationsCounter = 0;
```

Comment:
So, I have `window.restoreWindows` set to `all`. While I was debugging why the diagnostics showed but the decorations did not upon launching the editor, I found that I needed to add that bit on L177-186 in there. The log statement I put after `activeTextEditor.setDecorations()` was being logged repeatedly. I knew I needed to prevent it being called more than necessary.

I used a boolean at first and when I relaunched VSCode, the decorations from the restored file never appeared unless I toggled Show/Hide or clicked between tabs. I removed the boolean. They showed up again. So, I thought "I need a double boolean 😏 ", right, by using a counter? It started working. 

Apparently, it has to be called twice if you want them to display upon restoring a workspace with open files. It was very odd to witness.